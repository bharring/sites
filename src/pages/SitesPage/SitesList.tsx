import { RefresherEventDetail } from '@ionic/core';
import { IonItem, IonList, IonRefresher, IonRefresherContent } from '@ionic/react';

import { SitesListQuery } from '../../models/server/schema';
import SiteListItem from './SiteListItem';

const SitesList: React.FC<{
  data?: SitesListQuery;
  refetch: (variables?: any) => Promise<any>; // this is the problem with using generated code, the actual function signature of refetch is unwieldy
}> = ({ data, refetch }) => {
  const handleRefresh = async (e: CustomEvent<RefresherEventDetail>) => {
    try {
      await refetch();
    } catch {}
    e.detail.complete();
  };

  return (
    <>
      <IonRefresher slot="fixed" onIonRefresh={handleRefresh}>
        <IonRefresherContent></IonRefresherContent>
      </IonRefresher>

      <IonList>
        {data?.sitesList?.items &&
          data?.sitesList?.items.map((site) => (
            <IonItem button key={site?.id}>
              <SiteListItem site={site} />
            </IonItem>
          ))}
      </IonList>
    </>
  );
};

export default SitesList;
