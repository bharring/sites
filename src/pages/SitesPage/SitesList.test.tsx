import '@testing-library/jest-dom';

import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import SitesList from './SitesList';

const mockSprings = {
  id: '00000000-0000-0000-0000-000000000000',
  name: 'Bartons Springs Pool',
  address: {
    street: '2201 William Barton Dr',
    locality: 'Austin',
    region: 'TX',
    postalCode: '78746',
    country: 'US',
  },
  position: {
    latitude: 30.2641138,
    longitude: -97.7734661,
    altitude: null,
  },
};
const mockCapitol = {
  id: '11111111-1111-1111-1111-111111111111',
  name: 'Texas Capitol',
  address: {
    street: '1100 Congress Ave',
    locality: 'Austin',
    region: 'TX',
    postalCode: '78701',
    country: 'US',
  },
  position: {
    latitude: 30.2746698,
    longitude: -97.7425445,
    altitude: null,
  },
};
describe('<SitesList />', () => {
  afterEach(cleanup);

  test('matches snapshot', () => {
    const data = { sitesList: { items: [mockSprings, mockCapitol], nextToken: null, scannedCount: null } };

    const tree = renderer.create(<SitesList data={data} refetch={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });
  test('matches snapshot empty list', () => {
    const tree = renderer.create(<SitesList refetch={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });
});
