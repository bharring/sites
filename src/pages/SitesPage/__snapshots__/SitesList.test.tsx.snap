// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SitesList /> matches snapshot 1`] = `
Array [
  <ion-refresher
    slot="fixed"
  >
    <ion-refresher-content />
  </ion-refresher>,
  <ion-list>
    <ion-item
      button={true}
      onClick={[Function]}
    >
      <ion-label>
        <h2>
          Bartons Springs Pool
        </h2>
        <h3>
          2201 William Barton Dr, Austin, TX 78746
        </h3>
      </ion-label>
    </ion-item>
    <ion-item
      button={true}
      onClick={[Function]}
    >
      <ion-label>
        <h2>
          Texas Capitol
        </h2>
        <h3>
          1100 Congress Ave, Austin, TX 78701
        </h3>
      </ion-label>
    </ion-item>
  </ion-list>,
]
`;

exports[`<SitesList /> matches snapshot empty list 1`] = `
Array [
  <ion-refresher
    slot="fixed"
  >
    <ion-refresher-content />
  </ion-refresher>,
  <ion-list />,
]
`;
